core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -121,20 +127,31 @@ oppia.controller('FeedbackTab', [
         $scope.cancelReview = function() {
           $modalInstance.dismiss('cancel');
         };
+
+        $scope.suggestionValid = function() {
+          return (activeThread.status === 'open' &&
+            explorationData.data.version === suggestion.exploration_version);
+        }
       }]
     }).result.then(function(result) {
       threadDataService.resolveSuggestion(
-        $scope.activeThread.suggestion.suggestion.id, result.action);
-      // Update the status of the feedback thread.
-      var msg, status;
-      if (result.action === 'accept') {
-        msg = 'Suggestion accepted.';
-        status = 'fixed';
-      } else {
-        msg = 'Suggestion rejected.';
-        status = 'not_actionable';
-      }
-      $scope.addNewMessage($scope.activeThread.thread_id, msg, status);
+        $scope.activeThread.thread_id, result.action, result.commitMsg,
+        function(res) {
+          threadDataService.fetchThreads(function() {
+            $scope.setActiveThread($scope.activeThread.thread_id);
+          });
+          // Immediately update editor to reflect accepted suggestion.
+          if(result.action === 'accept') {
+            var suggestion = $scope.activeThread.suggestion;
+            var stateName = suggestion.state_name;
+            var state = angular.copy(explorationData.data.states[stateName]);
+            state.content[0].value = suggestion.state_content.value;
+            explorationStatesService.setState(stateName, state);
+            $rootScope.$broadcast('refreshStateEditor');
```

Comment:
I just realized an issue with this. What if the creator is editing an exploration, and then accepts a suggestion (before saving their draft)? After accepting the suggestion, they won't be able to save their existing changes because the version number will be wrong. (Could you confirm this is an issue, too?)

It's not great, but I think a workaround for now would be to prevent the creator from accepting suggestions until they've saved any drafts they currently have. Maybe we can display a message along the lines of "Please save your draft changes before accepting this suggestion." It's not ideal at all, unfortunately, but I think it will have to do for now.